contracts/interchain-token-service/src/contract.rs

```diff
@@ -329,380 +183,146 @@ impl AxelarExecutableInterface for InterchainTokenService {
-                    name: String::from_str(&env, &"Test Token"),
-                    symbol: String::from_str(&env, &"TST"),
-                    decimals: 18,
-                    minter: Some(bytes_from_hex(&env, "1234")),
-                }),
-            },
-        ];
-
-        for msg in msgs {
-            let encoded_hub_msg = assert_ok!(msg.clone().abi_encode(&env));
-            let (original_chain_name, inner_message) =
-                assert_ok!(env.as_contract(&client.address, || {
-                    InterchainTokenService::get_execute_params(
-                        &env,
-                        InterchainTokenService::its_hub_chain_name(&env),
-                        encoded_hub_msg,
-                    )
-                }));
-
-            if let HubMessage::ReceiveFromHub {
-                source_chain,
-                message,
-            } = msg
-            {
-                assert_eq!(original_chain_name, source_chain);
-                assert_eq!(inner_message, message);
-            }
+        Message::DeployInterchainToken(_) => {
+            // TODO
+            Ok(())
         }
     }
+}
 
-    #[test]
-    fn get_execute_params_p2p_message() {
-        let (env, client) = setup_env();
-        register_chains(&env, &client);
-
-        let msgs = vec![
-            Message::InterchainTransfer(InterchainTransfer {
-                token_id: BytesN::from_array(&env, &[255u8; 32]),
-                source_address: bytes_from_hex(&env, "4F4495243837681061C4743b74B3eEdf548D56A5"),
-                destination_address: bytes_from_hex(
-                    &env,
-                    "4F4495243837681061C4743b74B3eEdf548D56A5",
-                ),
-                amount: i128::MAX,
-                data: Some(bytes_from_hex(&env, "abcd")),
-            }),
-            Message::DeployInterchainToken(DeployInterchainToken {
-                token_id: BytesN::from_array(&env, &[1u8; 32]),
-                name: String::from_str(&env, &"Test Token"),
-                symbol: String::from_str(&env, &"TST"),
-                decimals: 18,
-                minter: Some(bytes_from_hex(&env, "1234")),
-            }),
-        ];
-
-        for msg in msgs {
-            let encoded_msg = assert_ok!(msg.clone().abi_encode(&env));
-            let (source_chain_name, message) = assert_ok!(env.as_contract(&client.address, || {
-                InterchainTokenService::get_execute_params(
-                    &env,
-                    String::from_str(&env, P2P_CHAIN),
-                    encoded_msg,
-                )
-            }));
-            assert_eq!(source_chain_name, String::from_str(&env, P2P_CHAIN));
-            assert_eq!(message, msg);
+fn get_execute_params(
+    env: &Env,
+    source_chain: String,
+    payload: &Bytes,
+) -> Result<(String, Message), ContractError> {
+    let message_type =
+        get_message_type(&payload.to_alloc_vec()).map_err(|_| ContractError::InvalidPayload)?;
+
```

Comment:
use `ensure!(message_type == ..)` so it doesn't need to be nested under a `match ...`